lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -119,8 +119,13 @@ sub details() {
         next unless $img->{screenshot};
         my $link = readlink($dir . "/" . $img->{screenshot});
         next unless $link;
-        $img->{md5_basename} = basename($link);
-        $img->{md5_dirname}  = basename(dirname($link));
+        my $base = basename($link);
+        my $dir  = dirname($link);
+        $dir =~ s,^.*/images/,,;
+        if ($dir ne 'ad') {    # avoid problem with ad block
+            $img->{md5_dirname}  = $dir;
+            $img->{md5_basename} = $base;
+        }
```

Comment:
The 'ad' directory will go away over time anyway - but I think that if is an acceptable workaround for confusions like yours. Migrating the images is another solution, but I will not do that soonish as it's a pain to test.